lib/new_relic/agent/error_collector.rb

```diff
@@ -250,26 +251,22 @@ def notice_segment_error(segment, exception, options = {})
 
       # See NewRelic::Agent.notice_error for options and commentary
       def notice_error(exception, options = {}, span_id = nil)
-        state = ::NewRelic::Agent::Tracer.state
-        transaction = state.current_transaction
-        status_code = transaction&.http_response_code
-
+        status_code = process_http_status_code(exception, options)
         return if skip_notice_error?(exception, status_code)
 
         tag_exception(exception)
 
-        if options[:expected] || @error_filter.expected?(exception, status_code)
+        state = ::NewRelic::Agent::Tracer.state
+        if options[:expected]
```

Comment:
Whats the reason for removing the || on the if? Is it because if it's expected it should have already returned on line 255?

Reply:
The bit of logic that was ORed (`||`) previously was moved to the new `process_http_status_code` method, which gets called before this line is reached. It doesn't lead to an early return, but it does lead to `expect: true` being set in the `options` hash itself. So now we can simplify the check here and only look for `expect: true` in the options hash.

Reply:
Ahh I see, that totally makes sense, thanks for clarifying!